Changes to release GitHub Pages by branch

mtwardawski opened the pull request:
Hi,

together with @mbfm we worked on another approach of getting our GitHub Pages build and released.

Our intention is to switch from a dedicated repository, which is `eclipse-opendut.github.io`, to a branch-based build and release process.
This changes were necessary since we want to integrate the building process into our Pipeline, which gets triggered in certain situation, i.e. a push to main or development branch.

For this to happen we extended our .jsonnet file:
  - adding three properties to repository `opendut`:

  - adding new environment to repository `opendut`
  
  - still kept the repository `eclipse-opendut.github.io` where it was, since we want to make a redirect to the new URL. Contents there will be removed, but a redirect will be added later.
  	
 
 I'm looking forward to hear your opinion and/or questions on our changes.
 
 Kind regards,
 Matthias Twardawski (Mercedes-Benz Tech Innovation)

mtwardawski committed:
Changed our structure in a way, to have our github pages build from a branch  from our opendut repository

mtwardawski committed:
Merge branch 'eclipse-opendut:main' into main

netomi commented:
I would strongly recommend to use the workflow approach to publish pages instead of the legacy one which relies on the fact that the content is pushed to a separate branch of the same repo.

If I understand you correctly, you are also willing to migrate the existing content in eclipse-opendut.github.io to the opendut repo, build the page and configure a redirect from eclipse-opendut.github.io to eclipse-opendut.github.io/opendut for that to work?

Tbh I have not tried that and do not know if this is possible considering that the GitHub Pages domain setup of GitHub is by convention, so maybe it can not be altered easily.

mtwardawski committed:
DOC: Issue-118 - Adjusted .jsonnet file to use GitHub workflow to build and deploy our GitHub Pages homepage. Added additional branch policies to match main and development branch and tags.

mtwardawski commented:
Hi @netomi,

thank you for your comment. I introduced the workflow based process in the file and added additional branch policies.

@mbfm and I talked about it and it seems to fit our needs. Also it looks good on my fork of opendut. 

About the re-direct, I think, we need to discuss this further in the team, if and how we want to proceed with that, since we also have a custom domain.

Kind regards,
Matthias

netomi commented:
If you already have a custom domain, the eclipse-opendut.github.io repo might not be needed at all and we can setup the custom domain in the pages configuration of the opendut repo itself. That would simplify the setup considerably.

netomi commented:
I added some project leads for approval. The change looks complete to me, so once its approved it can be merged and applied.

mtwardawski committed:
DOC: Issue-118 - deleted `canary` from branch_policies.

mtwardawski commented:
Hi @netomi,

thanks for your reply.
I did just one small change and deleted the `canary` tag from our branch_policies, since we don't want it to be re-published on that tag. 

Despite that, can you point me in the right direction, since I could not find any setting to be set to use our custom domain?
How can I  setup the custom domain in the pages configuration of our repo?

Have a nice weekend.
Best,
Matthias

netomi commented:
the custom domain setup can only be done manually via the Web UI. Once this is settled we can set this up separately.

mtwardawski commented:
@netomi 
alright thank you, we let you know, once this is finished :)

mbfm commented:
Hey @netomi, is it possible to get this merged or do we still need to do something?

netomi commented:
the author of the PR can merge it himself, I was not merging it myself as I was not sure if everything is done from your side.

mtwardawski commented:
Hi @netomi 
unfortunately I'm not able to merge it. It says that it's still waiting for the review of the code owner `eclipse-opendut/eclipsefdn-security`. Do you know, who is able to review and approve in behalf of this user?

netomi commented:
well you can merge it by adding a comment /otterdog merge as explained in a comment above.

mtwardawski commented:
/otterdog merge

mbfm commented:
Rolling out the documentation worked and the site is now deployed to <https://opendut.eclipse.dev/opendut/>.

@netomi You mentioned changing the custom domain in the GitHub settings. Is it possible to make `https://opendut.eclipse.dev/` point directly to `https://eclipse-opendut.github.io/opendut/`, so that the `/opendut` path is unnecessary?

Currently, the page is still also served directly at <https://opendut.eclipse.dev/>, because we still have the GitHub Pages repo at <https://github.com/eclipse-opendut/eclipse-opendut.github.io>.

netomi commented:
Sure, but can you please raise a ticket at the Eclipse HelpDesk about that?

mbfm commented:
I created a ticket here: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/4588
